### PR TITLE
Add a save and close function for prefabs

### PR DESCRIPTION
**Pull request by alexeykostin**
## What does this PR do?

Currently the standard prefab flow requires that you save and then separately close.

This PR simplifies the flow. The new menu item "Save and Close" introduced

SandCPrefab

## How was this PR tested?

Tested in the Editor, Window

### CODE REVIEW

**Review by carbonated-niko — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp [hunk 1]
```diff
@@ -865,7 +924,11 @@ namespace AzToolsFramework
             m_menuManagerInterface->AddActionToMenu(EditorIdentifiers::ViewportContextMenuIdentifier, "o3de.action.prefabs.detach", 20200);
             m_menuManagerInterface->AddActionToMenu(EditorIdentifiers::ViewportContextMenuIdentifier, "o3de.action.prefabs.instantiate", 20300);
             m_menuManagerInterface->AddActionToMenu(EditorIdentifiers::ViewportContextMenuIdentifier, "o3de.action.prefabs.procedural.instantiate", 20400);
+#if defined(CARBONATED)
+            m_menuManagerInterface->AddActionToMenu(EditorIdentifiers::ViewportContextMenuIdentifier, "o3de.action.prefabs.saveclose", 30100);
+#else
             m_menuManagerInterface->AddActionToMenu(EditorIdentifiers::ViewportContextMenuIdentifier, "o3de.action.prefabs.save", 30100);
+#endif
```
Comment (carbonated-niko) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp]: Oh actually - could you remove the else here so you have both options (save, save and close)

**Comment by carbonated-niko** [comment]
Additionally @alexeykostin please link the ticket in the PR in the future please

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp (hunk 1 above)
Comment (alexeykostin) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp]: @carbonated-niko Done
SandCPrefab2

Comment (carbonated-niko) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp]: Thanks! Lgtm

**Pull request merged**